From: Simon McVittie <smcv@debian.org>
Date: Mon, 19 Nov 2018 17:18:13 +0000
Subject: Document portal: Don't raise a critical when removing entries

We can store a NULL PermissionDbError (which is just an alias for
GVariant) in main_updates, as a "white-out" that indicates that an
entry has been removed. Callig g_variant_unref() on NULL is considered
to be a programming error.

Signed-off-by: Simon McVittie <smcv@debian.org>
Forwarded: https://github.com/flatpak/xdg-desktop-portal/pull/264
---
 document-portal/permission-db.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/document-portal/permission-db.c b/document-portal/permission-db.c
index 9ac2497..827a298 100644
--- a/document-portal/permission-db.c
+++ b/document-portal/permission-db.c
@@ -228,7 +228,7 @@ permission_db_init (PermissionDb *self)
 
   self->main_updates =
     g_hash_table_new_full (g_str_hash, g_str_equal,
-                           g_free, (GDestroyNotify) g_variant_unref);
+                           g_free, (GDestroyNotify) permission_db_entry_unref);
   self->app_additions =
     g_hash_table_new_full (g_str_hash, g_str_equal,
                            g_free, (GDestroyNotify) g_ptr_array_unref);
@@ -947,7 +947,8 @@ permission_db_entry_ref (PermissionDbEntry *entry)
 void
 permission_db_entry_unref (PermissionDbEntry *entry)
 {
-  g_variant_unref ((GVariant *) entry);
+  if (entry != NULL)
+    g_variant_unref ((GVariant *) entry);
 }
 
 /* Transfer: full */
